Contents/Code/youtube_dl_helper.py

```diff
@@ -35,19 +42,29 @@ def process_youtube(url):
     ...
     """
     youtube_dl_params = dict(
-        outmpl='%(id)s.%(ext)s',
-        youtube_include_dash_manifest=False,
-        username=Prefs['str_youtube_user'] if Prefs['str_youtube_user'] else None,
+        logger=plugin_logger,
+        outtmpl=u'%(id)s.%(ext)s',
         password=Prefs['str_youtube_passwd'] if Prefs['str_youtube_passwd'] else None,
+        socket_timeout=10,
+        username=Prefs['str_youtube_user'] if Prefs['str_youtube_user'] else None,
+        youtube_include_dash_manifest=False,
     )
 
     ydl = youtube_dl.YoutubeDL(params=youtube_dl_params)
 
     with ydl:
-        result = ydl.extract_info(
-            url=url,
-            download=False  # We just want to extract the info
-        )
+        try:
+            result = ydl.extract_info(
+                url=url,
+                download=False  # We just want to extract the info
+            )
+        except Exception as exc:
+            if isinstance(exc, youtube_dl.utils.ExtractorError) and exc.expected:
+                Log.Info('YDL returned YT error while downloading %s: %s' % (url, exc))
+            else:
+                Log.Exception('YDL returned an unexpected error while downloading %s: %s' % (url, exc))
+            return None
```

Comment:
The invalid test needs to be updated. https://github.com/zdimension/Themerr-plex/blob/eb08828d373ba51b20a5c3f1732f215a6a350d08/tests/unit/test_youtube_dl_helper.py#L22C1-L30C55

```diff
def test_process_youtube_invalid():
    # test invalid urls
    invalid_urls = [
        'https://www.youtube.com/watch?v=notavideoid',
        'https://blahblahblah',
    ]
    for url in invalid_urls:
-        with pytest.raises(DownloadError):
-            youtube_dl_helper.process_youtube(url=url)
+        audio_url = youtube_dl_helper.process_youtube(url=url)
+        assert audio_url is None
```